### PR TITLE
lib/streamaggr: skip empty aggregators

### PR DESCRIPTION
**Pull request by hagen1778**
Prevent excessive resource usage when stream aggregation config file contains no matchers by prevent pushing data into Aggregators object. Before this change a lot of extra work was invoked without reason.

### CODE REVIEW

File: app/vmagent/remotewrite/remotewrite.go [hunk 1]
```diff
@@ -492,7 +492,7 @@ func tryPush(at *auth.Token, wr *prompbmarshal.WriteRequest, forceDropSamplesOnF
 		}
 		sortLabelsIfNeeded(tssBlock)
 		tssBlock = limitSeriesCardinality(tssBlock)
-		if sas != nil {
+		if sas.IsEnabled() {
```
Comment (AndrewChubatiuk) [inline app/vmagent/remotewrite/remotewrite.go:495]: @hagen1778 don't we need the same for global aggregator?

Comment (AndrewChubatiuk) [reply app/vmagent/remotewrite/remotewrite.go:495]: or can we return nil for loadAggregatorsFromData if len(a.as) == nil?

Comment (hagen1778) [reply app/vmagent/remotewrite/remotewrite.go:495]: Could you please point me to the specific code line or send a follow-up commit?

Comment (AndrewChubatiuk) [reply app/vmagent/remotewrite/remotewrite.go:495]: sorry, just seen a variable name in context, expected other variable name for global aggregator

**Review by AndrewChubatiuk — APPROVED** [review]

**Pull request merged**